src/main/java/coupon/repository/CouponInMemoryRepository.java

```diff
@@ -0,0 +1,43 @@
+package coupon.repository;
+
+import coupon.domain.Coupon;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.redis.serializer.Jackson2JsonRedisSerializer;
+import org.springframework.stereotype.Repository;
+import redis.clients.jedis.JedisPooled;
+import java.util.NoSuchElementException;
+
+@Repository
+@RequiredArgsConstructor
+public class CouponInMemoryRepository {
+
+    private static final Jackson2JsonRedisSerializer<Coupon> serializer = new Jackson2JsonRedisSerializer<>(Coupon.class);
```

Comment:
```suggestion
        private static final RedisSerializer<Coupon> serializer = new Jackson2JsonRedisSerializer<>(Coupon.class);
```
제안) 추상화 인터페이스에 의존하는 건 어떤가요?

Reply:
RedisSerializer가 CouponInMemoryRepository에 있는것이 자연스럽지 않게 느껴져요
RedisConfig에서 빈으로 등록해 사용할 수 있을 것 같아요!

Reply:
1. 인터페이스로 변경했습니다!

2. 저는 Coupon에 대한 직렬화 객체이기 때문에 CouponInMemoryRepository에 있어도 괜찮다고 생각합니다!
빈으로 등록해야 하는 이유도 잘 모르겠습니다.